src/extension.ts

```diff
@@ -424,3 +423,33 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
         context.subscriptions.push(createClient(dirToAnalyze).start());
     }
 }
+
+// Search all the workspace folders and return the first that contains .phan/config.php
+// This will return the folder name in an array as required, and just only one folder,
+// the first met. We can easy modify the function to add all valid workspace folders to the array.
+function findValidProjectDirectories(): string[] {
+    // Get the fsPath(file system path) of all workspace folders.
+    const VSCodeFolders = vscode.workspace.workspaceFolders;
+    if (!VSCodeFolders) {
+        return [];
+    }
+    let workingFolders = VSCodeFolders.map(function (obj) {
+        if (('uri' in obj) && ('fsPath' in obj.uri)) {
+            return obj.uri.fsPath;
+        }
+        return '';
+    });
+
+    workingFolders = workingFolders.filter(function (folderPath) {
+        return pathContainsPhanFolderAndConfig(folderPath);
+    });
+
+    return workingFolders;
+}
+
+// Whether or not a path contains the ".phan" folder alongside with a config.php file.
```

Comment:
`// Returns true if a path contains ".phan/config.php"`

"alongside with" could be read as containing both `$PROJECT/.phan` and `$PROJECT/config.php`

Reply:
Ok